programs/ssl/ssl_client2.c

```diff
@@ -2044,6 +2045,8 @@ int main( int argc, char *argv[] )
 
         mbedtls_printf( "  . Reconnecting with saved session..." );
 
+        memset( peer_crt_info, 0, sizeof( peer_crt_info ) );
```

Comment:
Thanks for adding this. However, there is another code path that reconnects (7b in the comment) which I think need this as well. (Found by looking for other calls to `mbedtls_ssl_session_reset()`.)

Reply:
Sorry for that! Fixed.